pkg/webhook/validate_service.go

```diff
@@ -46,10 +47,20 @@ const (
 func ValidateRevisionTemplate(ctx context.Context, uns *unstructured.Unstructured) error {
 	content := uns.UnstructuredContent()
 
+	var mode DryRunMode
+	features := config.FromContextOrDefaults(ctx).Features
+	switch features.PodSpecDryRun {
+	case config.Enabled:
+		mode = DryRunEnabled
+	case config.Disabled:
+		mode = DryRunMode("")
```

Comment:
this might be simpler

```suggestion
		return nil
```

Reply:
"cannot use nil as type DryRunMode in assignment"